drivers/interrupt_controller/CMakeLists.txt

```diff
@@ -23,3 +23,6 @@ zephyr_library_sources_ifdef(CONFIG_SHARED_IRQ              intc_shared_irq.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_ESP32               intc_esp32.c)
 zephyr_library_sources_ifdef(CONFIG_SWERV_PIC               intc_swerv_pic.c)
 zephyr_library_sources_ifdef(CONFIG_VEXRISCV_LITEX_IRQ      intc_vexriscv_litex.c)
+
+# Silence CMake warning when no interrupt controller driver is enabled
+zephyr_library_sources(${ZEPHYR_BASE}/misc/empty_file.c)
```

Comment:
Nothing wrong, but I kind of feel sad whenever we need to add `empty_file.c` to libraries.
It just feels wrong.

we could change this file into something like this.
```
list_append_ifdef(CONFIG_ARCV2_INTERRUPT_UNIT file_list intc_arcv2_irq_unit.c)
list_append_ifdef(CONFIG_CAVS_ICTL            file_list intc_cavs.c)
list_append_ifdef(CONFIG_DW_ICTL              file_list intc_dw.c)
....

if(file_list)
  zephyr_library()
  zephyr_library_sources(${file_list})
endif()
```

Although, I also like the current structure as it conforms to rest of our CMakeLists files.

Will approve both versions, but just wanted to state the alternative for others to pitch in.

Reply:
+1 to using a file_list.

-1 to empty_file.c

One approach slows down the build and the other does not.

In hindsight Interrupt controllers should have been a menuconfig instead of a menu. Like the other drivers.
But introducing this now would break compatibility.

Another approach would be for each interrupt controller to select a generic promptless INTERRUPT_CONTROLLER kconfig, but this is not worth the effort. If there has been no use for a generic Kconfig option yet then there probably never will be.